File: card-core/src/main/java/com/jean202/cardmizer/core/domain/PerformanceTier.java
package com.jean202.cardmizer.core.domain;

import com.jean202.cardmizer.common.Money;
import java.util.Objects;

public record PerformanceTier(String code, Money targetAmount, String benefitSummary) {
    public PerformanceTier {
        if (code == null || code.isBlank()) {
            throw new IllegalArgumentException("Tier code must not be blank");
        }
        targetAmount = Objects.requireNonNull(targetAmount, "targetAmount must not be null");
        if (benefitSummary == null || benefitSummary.isBlank()) {
            throw new IllegalArgumentException("Benefit summary must not be blank");
        }
    }
}

File: card-core/src/main/kotlin/com/jean202/cardmizer/core/domain/PerformanceTier.kt
package com.jean202.cardmizer.core.domain

import com.jean202.cardmizer.common.Money

data class PerformanceTier(
    val code: String,
    val targetAmount: Money,
    val benefitSummary: String,
) {
    init {
        require(code.isNotBlank()) { "Tier code must not be blank" }
        require(benefitSummary.isNotBlank()) { "Benefit summary must not be blank" }
    }
}
